lib/CStack.cpp

```diff
@@ -102,10 +102,10 @@ si32 CStack::magicResistance() const
 		if(one->unitOwner() == owner)
 			vstd::amax(auraBonus, one->valOfBonuses(Bonus::SPELL_RESISTANCE_AURA)); //max value
 	}
-	magicResistance += auraBonus;
+	magicResistance *= 100.0/auraBonus;
 	vstd::amin(magicResistance, 100);
 
-	return magicResistance;
+	return static_cast<si32>(magicResistance);

```

Comment:
Are you sure that formula is correct?
Assuming magicResistance = 40 and auraBonus = 20
This line will give 40 * 100 / 20 = 200

Additive bonus as before should result in 60
Presumably correct multiplicative bonus should give (100 - (100 - 40) * (100 - 20)) = 52

Also, I would rather keep formula as simple as possible to make code more clear, even if this leads to few extra math operations.

Reply:
@IvanSavenko thanks for catching. Updated formula to 100/(100-auraBonus), and added abetween for auraBonus.
What I need to do to keep math more simple? Separate auraBonus calculations altogether?

Reply:
I will fix it more ASAP.

Reply:
So, it should work like this:
```100 - ((100 - magicResistance) * (100 - auraBonus)/100.0)```

Because this is what FizMig says: 
```
 Шанс срабатывания заклинания и шанс блокирования – не одно и то же.
Например, если рядом с единорогом находится Кентавр, то шанс срабатывания вражеского Проклятья по нему снижается со 100% до 80%. А если рядом с
единорогом находится Гном, то шанс срабатывания снижается с 80% до 80%*0,8=64%, т.е. сопротивление Гнома становится равным 36%.
```

Reply:
Did it as FizMig.